### PR TITLE
fix: disable scheduler on errors

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash/issues/10297
Closes: #10298 //do not cascade on these errors 

### Description:

Disable scheduler / gsheet for these 3 error types: 

- Worker worker-f18d1f5ab12d46f2ad failed job 453905 (uploadGsheets). Error: Error: invalid_grant https://lightdash.sentry.io/issues/4150573551/?project=5959292&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=issue-stream&statsPeriod=24h&stream_index=0
- Worker worker-bb3f4229725d34adf9 failed job 1922 (handleScheduledDelivery). Error: Could not find slack installation https://lightdash.sentry.io/issues/4202827650/?project=5959292&query=is%3Aunresolved+Worker&referrer=issue-stream&statsPeriod=24h&stream_index=2
- Worker worker-f18d1f5ab12d46f2ad failed job 453903 (uploadGsheets). Error: Error: Requested entity was not found. https://lightdash.sentry.io/issues/4150573551/events/442eaec39a504730914af13701108bd3/?project=5959292&query=is%3Aunresolved+Worker&referrer=previous-event&statsPeriod=24h&stream_index=0

Screenshot from 2024-06-05 09-58-58

Note: the disabled status is not re-fetch in the frontend, this is expected and out of scope for this ticket. 

[Screencast from 05-06-24 10:11:36.webm](https://github.com/lightdash/lightdash/assets/1983672/bfd46a54-714b-4715-9540-2755187a381b)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved user session handling in the scheduler to ensure correct user data assignment.

### CODE REVIEW

File: packages/backend/src/scheduler/SchedulerTask.ts [hunk 1]
```diff
@@ -1444,6 +1446,19 @@ export default class SchedulerTask {
                 status: SchedulerJobStatus.COMPLETED,
             });
         } catch (e) {
+            if (
+                `${e}`.includes('invalid_grant') ||
+                `${e}`.includes('Requested entity was not found')
```
Comment (joaoviana) [inline packages/backend/src/scheduler/SchedulerTask.ts:1451]: what about `Could not find slack installation` ?

File: packages/backend/src/scheduler/SchedulerTask.ts [hunk 2]
```diff
@@ -1444,6 +1446,19 @@ export default class SchedulerTask {
...
+            ) {
+                console.warn(
+                    `Disabling scheduler with non-retryable error: ${e}`,
+                );
+                await this.schedulerService.setSchedulerEnabled(
```
Comment (joaoviana) [inline packages/backend/src/scheduler/SchedulerTask.ts:1456]: after this runs, should we still cascade the error?

**Review by joaoviana — COMMENTED** [review]
2 Qs!

File: packages/backend/src/scheduler/SchedulerTask.ts (hunk 2 above)
Comment (rephus) [reply packages/backend/src/scheduler/SchedulerTask.ts:1456]: Good question, probably not, this is what https://github.com/lightdash/lightdash/issues/10298 is supposed to achieve, right ?

File: packages/backend/src/scheduler/SchedulerTask.ts (hunk 1 above)
Comment (rephus) [reply packages/backend/src/scheduler/SchedulerTask.ts:1451]: Good question, I forgot to push that code... :facepalm:

File: packages/backend/src/scheduler/SchedulerTask.ts (hunk 2 above)
Comment (joaoviana) [reply packages/backend/src/scheduler/SchedulerTask.ts:1470]: `Perhaps we should not say jobs have "failed" for some types of errors`

Great!

**Review by joaoviana — APPROVED** [review]

**Pull request merged**